tools/bqtag/bqtag/bqtag.py

```diff
@@ -0,0 +1,800 @@
+import google.auth
+
+__all__ = ["BQTableView"]
+
+#############################################################################
+# Create Logger
+#############################################################################
+
+FORMAT = "%(asctime)-15s %(levelname)s %(message)s"
+
+def get_logger(name, fmt):
+    """
+    Creates a Logger that logs to stdout
+    :param name: name of the logger
+    :param fmt: format string for log messages
+    :return: Logger
+    """
+    handler = logging.StreamHandler(sys.stdout)
+    handler.setLevel(logging.DEBUG)
+    handler.setFormatter(logging.Formatter(fmt))
+    log = logging.getLogger(name)
+    log.setLevel(logging.DEBUG)
+    log.addHandler(handler)
+    return log
+
+LOGGER = get_logger("bqtag", FORMAT)
+
+
+#############################################################################
+# Classes to create tree structure to create SQL query for nested columns
+#############################################################################
+
+
+class ColumnNode:
+
+    """
+    Class to represent a BQ column in a tree
+    """
+
+    def __init__(self, name="", parent="", column_json = None, children = None):
+        self.name = name
+        self.json = column_json
+        self.children = children
+        self.parent = parent
+        self.mode = ""
+
+class ColumnTree:
+
+    """
+    Class to represent a tree of columns for a BQ table
+    """
+
+    def __init__(self):
+        self.root = ColumnNode(name="Root",
+                               column_json=copy.copy({}),
+                               children=copy.copy({}))
+
+    def add_node(self, node_json: dict()):
+
+        cur_node = self.root
+
+        path = node_json["name"].split(".")
+        parents = path[:-1]
+        node_name = path[-1]
+
+        for parent in parents:
+
+            if parent in cur_node.children:
+                cur_node = cur_node.children[parent]
+            else:
+                new_node = ColumnNode(name=parent, parent=cur_node.name,
+                                      column_json=copy.copy({}),
+                                      children=copy.copy({}))
+
+                cur_node.children[parent] = new_node
+                cur_node = new_node
+
+        if node_name in cur_node.children:
+            cur_node.children[node_name].json = node_json
+            cur_node.mode = node_json["parent_mode"]
+        else:
+            cur_node.children[node_name] = ColumnNode(
+                                                name=node_name,
+                                                parent=cur_node.name,
+                                                column_json=copy.copy(node_json),
+                                                children=copy.copy({})
+                                            )
+            cur_node.mode = node_json["parent_mode"]
+
+
+    def generate_query(self):
+
+        """
+        Main function to generate SQL query for columns stored in tree
```

Comment:
Dot at the end of single line docstring (PEP 257)

Reply:
changed to
```
    def generate_query(self):
        """
        Main function to generate SQL query for columns stored in tree.
        """
```